services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +43,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  if(thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+
+  if(xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  error_code_t errCode;
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK};
+  
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  error_code_t errCode = ERR_CODE_SUCCESS;
+  thermal_mgr_event_t event;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+      if (event.type != THERMAL_MGR_EVENT_MEASURE_TEMP_CMD 
+       && event.type != THERMAL_MGR_EVENT_ISR_OVERTEMP_CHECK){
+        LOG_ERROR_CODE(ERR_CODE_INVALID_STATE);
+        return;
+      }
+      
+      float temp;
+      LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp));
+
+      if (errCode != ERR_CODE_SUCCESS) return;
```

Comment:
You never want to return from a freeRTOS function, it will probably crash the program, so replace all return statements with continue